[PKG-3090] u8darts 0.26.0 ❄️

lorepirri opened the pull request:
[PKG-3090] darts 0.26.0 ❄️ 

- destination channel: **Snowflake**

- upstream: https://github.com/unit8co/darts/tree/2f7c9dbb0e9e281cb1d74d609748a2b84fcec969 (the tagged 0.26.0 is 1 commit older https://github.com/unit8co/darts/tree/0.26.0 thus the patch number `002`)
- pypi: https://pypi.org/project/u8darts/
- pypi: https://pypi.org/project/darts/
- conda-forge: https://github.com/conda-forge/u8darts-feedstock

- dependencies:
    - https://github.com/unit8co/darts/blob/0.26.0/setup.py
    - https://github.com/unit8co/darts/tree/0.26.0/requirements
    - https://github.com/unit8co/darts/blob/0.26.0/pyproject.toml
    - https://github.com/unit8co/darts/blob/0.26.0/setup_u8darts.py

- patch `001_fix-resample-restore-loffset.patch`:
    - For `py38` and `py39` one test fails after the change in this PR: https://github.com/unit8co/darts/pull/1932/files
        in the file `darts/tests/test_timeseries.py`.
    - The patch restores one line reverting to the previous way of invoking the `resample` function.
- patch `002_add-latest-commit-0-26-0.patch`:
    - the tagged 0.26.0 is 1 commit older https://github.com/unit8co/darts/tree/0.26.0 thus the patch

- The package name is `u8darts`. It’s a multi-output package with the following outputs (see this [comment](https://github.com/AnacondaRecipes/u8darts-feedstock/pull/1#issuecomment-1774813629)):
        
        1. `u8darts`
        1. `u8darts-notorch`
        1. `u8darts-torch`
        1. `u8darts-all`


- depends on: AnacondaRecipes/nfoursid-feedstock#2

[PKG-3090]: https://anaconda.atlassian.net/browse/PKG-3090?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

lorepirri committed:
first commit

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
ERROR: clone/recipe/meta.yaml:17: outputs_not_unique: output "u8darts": Output name is not unique
Errors were found

</details>

lorepirri committed:
change name to u8darts-suite

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

lorepirri committed:
fix script

lorepirri committed:
fix name variable

lorepirri committed:
use scripts, and python

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

lorepirri committed:
add upload channel, tmp channels

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
ERROR: clone/recipe/meta.yaml:32: version_constraints_missing_whitespace: output "u8darts": Packages and their version constraints must be space separated
Errors were found

</details>

lorepirri committed:
fix linter

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

lorepirri committed:
remove staging channel

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

lorepirri commented:
> Linter check found the following problems:
> The following problems have been found:
> INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script. All checks OK

why this?

lorepirri committed:
patch one test

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
ERROR: clone/recipe/meta.yaml:25: patch_unnecessary: patch should not be in build when source/patches is not set
Errors were found

</details>

lorepirri committed:
skip s390x

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
ERROR: clone/recipe/meta.yaml:26: patch_unnecessary: patch should not be in build when source/patches is not set
Errors were found

</details>

lorepirri committed:
patch all

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
ERROR: clone/recipe/meta.yaml:26: patch_unnecessary: patch should not be in build when source/patches is not set
Errors were found

</details>

lorepirri committed:
move patches to the right place

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

lorepirri committed:
patch for py38 and py39

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

lorepirri committed:
make tests pass for arm64

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

lorepirri committed:
patch all

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

lorepirri committed:
patch: fix test dataset loaders

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
All checks OK

</details>

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -204,21 +237,31 @@ outputs:
+        - darts
+      commands:
+        - pip check
+        - pytest darts/tests
```
```suggestion
```
Run tests only once

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -156,23 +187,25 @@ outputs:
+        - darts
+      commands:
+        - pip check
+        - pytest darts/tests
```
```suggestion
```
Run tests only once

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -111,23 +140,25 @@ outputs:
+        - darts
+      commands:
+        - pip check
+        - pytest darts/tests
```
```suggestion
```

lorepirri committed:
skip tests for other outputs, temp skip all tests

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
ERROR: clone/recipe/meta.yaml:1: missing_test_requirement_pip: pip is required in the test requirements.
Errors were found

</details>

skupr-anaconda commented:
prophet-1.1.4 is available for py38

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
ERROR: clone/recipe/meta.yaml:1: missing_test_requirement_pip: pip is required in the test requirements.
Errors were found

</details>

skupr-anaconda commented:
**u8darts-all** has this error:
```
ModuleNotFoundError: No module named 'darts'

Tests failed for u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2 - moving package to /croot/broken

WARNING:conda_build.build:Tests failed for u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2 - moving package to /croot/broken

TESTS FAILED: u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2
```

lorepirri committed:
remove other test imports

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts": Output is missing script.
ERROR: clone/recipe/meta.yaml:1: missing_test_requirement_pip: pip is required in the test requirements.
ERROR: clone/recipe/meta.yaml:125: missing_imports_or_run_test_py: output "u8darts-notorch": Python packages require imports or a python test file in tests.
ERROR: clone/recipe/meta.yaml:144: missing_imports_or_run_test_py: output "u8darts-torch": Python packages require imports or a python test file in tests.
ERROR: clone/recipe/meta.yaml:165: missing_imports_or_run_test_py: output "u8darts-all": Python packages require imports or a python test file in tests.
Errors were found

</details>

lorepirri commented:
> **u8darts-all** has this error:
> 
> ```
> ModuleNotFoundError: No module named 'darts'
> 
> Tests failed for u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2 - moving package to /croot/broken
> 
> WARNING:conda_build.build:Tests failed for u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2 - moving package to /croot/broken
> 
> TESTS FAILED: u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2
> ```

I think it was part of the imports. I removed all of them (I kept only the main ones)

skupr-anaconda commented:
> > **u8darts-all** has this error:
> > ```
> > ModuleNotFoundError: No module named 'darts'
> > 
> > Tests failed for u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2 - moving package to /croot/broken
> > 
> > WARNING:conda_build.build:Tests failed for u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2 - moving package to /croot/broken
> > 
> > TESTS FAILED: u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2
> > ```
> 
> I think it was part of the imports. I removed all of them (I kept only the main ones)

I would not remove imports because the basic one is still required like we did for modin https://github.com/AnacondaRecipes/modin-feedstock/blob/477b7f171757cc331f4d15002cf8e76fa9bc7f76/recipe/meta.yaml#L56C1-L56C1

lorepirri committed:
revert remove other test imports

lorepirri commented:
> > > **u8darts-all** has this error:
> > > ```
> > > ModuleNotFoundError: No module named 'darts'
> > > 
> > > Tests failed for u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2 - moving package to /croot/broken
> > > 
> > > WARNING:conda_build.build:Tests failed for u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2 - moving package to /croot/broken
> > > 
> > > TESTS FAILED: u8darts-all-0.26.0-py311hd43f75c_0.tar.bz2
> > > ```
> > 
> > 
> > I think it was part of the imports. I removed all of them (I kept only the main ones)
> 
> I would not remove imports because the basic one is still required like we did for modin https://github.com/AnacondaRecipes/modin-feedstock/blob/477b7f171757cc331f4d15002cf8e76fa9bc7f76/recipe/meta.yaml#L56C1-L56C1

Ok, restored, also the restored `requires: pip`

lorepirri committed:
Merge branch 'PKG-3090-udarts-8-all-0-26-0' of https://github.com/AnacondaRecipes/u8darts-feedstock into PKG-3090-udarts-8-all-0-26-0

lorepirri committed:
add dart as main package

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "darts": Output is missing script.
All checks OK

</details>

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -66,24 +94,69 @@ outputs:
+  - name: u8darts
+    skip: true  # [py<38 or s390x]
+    requirements:
+      build:
+        - patch     # [not win]
+        - m2-patch  # [win]
```
It's redundant for this one and all outputs below
```suggestion
      build:
        - patch     # [not win]
        - m2-patch  # [win]
```
It's redundant for this one and all outputs below

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -66,24 +94,69 @@ outputs:
+        - m2-patch  # [win]
+      host:
+        - python
+        - pip
+        - setuptools
+        - wheel
```
```suggestion
```
It's redundant for this one and all outputs below

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -66,24 +94,69 @@ outputs:
+        - setuptools
+        - wheel
+      run:
+        - {{ pin_subpackage("darts", exact=True) }}
```
```suggestion
        - python
        - {{ pin_subpackage("darts", exact=True) }}
```

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -66,24 +94,69 @@ outputs:
+    test:
+      imports:
+        - darts
+        - darts.ad
+        - darts.ad.aggregators
+        - darts.ad.anomaly_model
+        - darts.ad.detectors
+        - darts.ad.scorers
+        - darts.dataprocessing
+        - darts.dataprocessing.dtw
+        - darts.dataprocessing.encoders
+        - darts.dataprocessing.transformers
+        - darts.datasets
+        - darts.explainability
+        - darts.metrics
+        - darts.models
+        - darts.tests
+        - darts.tests.dataprocessing
+        - darts.tests.dataprocessing.transformers
+        - darts.tests.utils
+        - darts.utils
+        - darts.utils.data
+        - darts.utils.historical_forecasts
```
```suggestion
```
import darts will be enough

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -66,24 +94,69 @@ outputs:
       host:
-        - python >=3.8
+        - python
+        - pip
+        - setuptools
+        - wheel
```
```suggestion
```
Redundant

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -111,24 +184,22 @@ outputs:
       host:
-        - python >=3.8
+        - python
+        - pip
+        - setuptools
+        - wheel
```
```suggestion
```
Redundant

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -156,24 +227,22 @@ outputs:
       host:
-        - python >=3.8
+        - python
+        - pip
+        - setuptools
+        - wheel
```
```suggestion
```
Redundant

lorepirri committed:
Apply suggestions from code review

Co-authored-by: Serhii Kupriienko <79282962+skupr-anaconda@users.noreply.github.com>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "darts": Output is missing script.
All checks OK

</details>

lorepirri committed:
only two subpackages, the rest commented out

lorepirri committed:
rename to `darts-packages`

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "darts": Output is missing script.
All checks OK

</details>

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -1,39 +1,61 @@
+    # For some reason the selectors [py38 or py39] and [py<310] do
+    # not work, so the patch is applied for all Python versions.
+    #
+    # In this patch we restore one line reverting to the
```
These comments should be put into the patches IMO.

lorepirri replied:
done

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -43,6 +65,11 @@ outputs:
+        # some tests fail if this pinning is not explicit
+        # because the version 2.8.3+snowflake will be picked
+        # up and apparently it's causing those tests fail
+        - python-dateutil 2.8.2
```
This will cause conflicts and will also cause `2.8.3+snowflake` to never be picked up. It would be interesting to see which tests are failing and check with Snowflake if they are ok with this or not.

lorepirri replied:
I commented it out right now. It's the outcome of many trial-and-error attempts, and it might not be important. Eventually, I will remove it later.

JeanChristopheMorinPerso commented on recipe/patches/001_fix-resample-restore-loffset.patch:
Did we develop this patch or was it taken from the upstream repo?

lorepirri replied:
yes I did, to revert a change in the PR: https://github.com/unit8co/darts/pull/1932/files

lorepirri committed:
move comments to patches

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "darts": Output is missing script.
All checks OK

</details>

lorepirri committed:
remove python-dateutil 2.8.2 pinning

lorepirri committed:
some formatting

lorepirri committed:
add requirements similarly to modin

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "darts": Output is missing script.
All checks OK

</details>

lorepirri commented:
Right now the main problem is that with `ModuleNotFoundError: No module named 'darts'` on a simple `import darts` in the `tests` section, it seems that the recipe is not well formatted, for some reason. I got blind to the problem after many trials. @JeanChristopheMorinPerso any idea? Right now I commented out all of the subpackages except the first two.

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -43,6 +63,11 @@ outputs:
+        # some tests fail if this pinning is not explicit
+        # because the version 2.8.3+snowflake will be picked
+        # up and apparently it's causing those tests fail
+        # - python-dateutil 2.8.2
```
I'm still not a fan of this. Have we asked Snowflake's opinion? They are the one who provided us the patch to python-dateutil. If their patch breaks something, then we need to tell them.

JeanChristopheMorinPerso committed:
Fix problem with import

lorepirri committed:
try: add feedstock-name

lorepirri committed:
add script for subpackage

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -66,163 +86,173 @@ outputs:
-    build:
-      noarch: python
+  - name: u8darts
+    script: base_build.sh  # [unix]
+    script: base_build.bat # [win]
```
```suggestion
```
You don't need these since you have `{{ pin_subpackage("darts", exact=True) }}` in the runtime dependencies.

lorepirri replied:
ok removed

lorepirri committed:
remove additional scripts

Co-authored-by: Jean-Christophe Morin <38703886+JeanChristopheMorinPerso@users.noreply.github.com>

skupr-anaconda commented:
Looks like one test fails on osx-64 because of this issue https://github.com/unit8co/darts/issues/195

skupr-anaconda commented:
On linux-64 only one test `test_ok_dataset[dataset_config19]` fails:
`E           darts.datasets.dataset_loaders.DatasetLoadingException: Could not download the dataset. Reason:BadZipFile('File is not a zip file')`

skupr-anaconda commented:
Errors on win64 for different reasons (one of them `PermissionError: [Errno 13] Permission denied: 'C:\\cygwin64\\tmp\\tmpq6u54fvw'`):
```

=========================== short test summary info ===========================
FAILED darts/tests/test_timeseries.py::TestTimeSeries::test_from_csv - Permis...
FAILED darts/tests/test_timeseries.py::TestTimeSeriesFromDataFrame::test_time_col_convert_string_integers
FAILED darts/tests/test_timeseries.py::TestTimeSeriesFromDataFrame::test_time_col_convert_integers
FAILED darts/tests/test_timeseries_multivariate.py::TestTimeSeries::test_from_csv
FAILED darts/tests/datasets/test_dataset_loaders.py::TestDatasetLoader::test_ok_dataset[dataset_config20]
```

I would skip some tests

JeanChristopheMorinPerso commented:
For `FAILED darts/tests/datasets/test_dataset_loaders.py::TestDatasetLoader::test_ok_dataset[dataset_config20]
` we could probaly just `-k 'not slow'` (deselect the "slow" pytest marker.

lorepirri committed:
skip tests marked as slow

lorepirri committed:
enable build subpackages

lorepirri committed:
remove patch for retry slow download datasets

lorepirri committed:
remove skip from outputs

lorepirri committed:
remove addional imports

skupr-anaconda committed:
Use pip check only for darts-all, rename sub-packages

skupr-anaconda commented:
I made the next changes:
- renamed sub-packages as `darts-` (_not_ **u8**darts-)
- Use `pip check` only for `darts-all`, to avoid failing before we build darts-torch because the setup.py requires base and torch requirements by default https://github.com/unit8co/darts/blob/9982a503d3e55ea3268db36c99e0e6c91b0be397/setup.py#L15

skupr-anaconda commented:
`Fatal Python error: Segmentation fault` on `osx-arm64`:
`/var/folders/nz/j6p8yfhx1mv_0grj5xl4650h0000gp/T/abs_46vcpl684j/croot/darts-packages_1697788725438/test_tmp/run_test.sh: line 7: 42696 Segmentation fault: 11  pytest -vvv -s -m "not slow" darts/tests`

First I though this can help https://github.com/AnacondaRecipes/pytorch-forecasting-feedstock/blob/8ec157506a5a097984c57c796fe6e503f0a9d155/recipe/meta.yaml#L25-L29 but it's related to **osx-64** **not** to `osx-arm64` as we have now 

UPD. Another run has a different error message:

```
OMP: Error #13: Assertion failure at kmp_dispatch.cpp(1298).

OMP: Hint Please submit a bug report with this message, compile and run commands used, and machine configuration info including native compiler and operating system versions. Faster response will be obtained by including all program sources. For information on submitting this issue, please see https://bugs.llvm.org/.

Fatal Python error: Aborted

...

/var/folders/k1/30mswbxs7r1g6zwn8y4fyt500000gp/T/abs_58vzsra4lw/croot/darts-packages_1697793772254/test_tmp/run_test.sh: line 7: 90335 Abort trap: 6           pytest darts/tests -vv -s -m "not slow" -k "not test_models_performance[config21]"
```

skupr-anaconda committed:
Restructure the recipe and skip tests:

- Rename subpackages

- Add new subpackages u8darts, u8darts-all

- Add comments about dummy tests for test/imports

- Skip failing tests on osx and win (they looks like the upstream bugs)

- Remove an unused comment about python-dateutils

- Move pip check to darts-all

- Add comments about skipping pip check for the darts subpackage

skupr-anaconda committed:
Fix a comment

skupr-anaconda committed:
Use jinja2 variable for subpackage names

lorepirri committed:
add patch to bump version number [skip ci]

lorepirri committed:
skip test_models_accuracy_univariate on osx [skip ci]

lorepirri committed:
temp: build only osx

lorepirri committed:
skip test_models_accuracy_multivariate on osx

lorepirri committed:
skip tests that fail for osx

lorepirri committed:
skip test_autoregressive_timeseries for win, trigger osx-arm64

lorepirri committed:
trigger

lorepirri committed:
skip test_models_runnability on osx-arm64

lorepirri committed:
skip test_models_performance on osx-arm64

lorepirri committed:
typo

lorepirri committed:
skip test_encoders for osx-arm64

lorepirri committed:
osx and win works, skip ppc64le, skip pip_checkl for win, final build

lorepirri commented:
for `win` the `darts-all` requires `xgboost` which is there but not detected by `pip`, removing `pip check`
```
                   INFO   import: 'darts'                                                                                                                                                                         
                   INFO   import: 'darts'                                                                                                                                                                         
                   INFO   darts 0.26.0 requires xgboost, which is not installed.                                                                                                                                  
                   INFO   TESTS FAILED: darts-all-0.26.0-py310haa95532_0.tar.bz2                                                                                                                                  
```

lorepirri committed:
typo, skip test_exogenous_variables_support for osx-arm64

lorepirri commented:
remember me to rename the repository (and the submodule) to `darts-feedstock`

JeanChristopheMorinPerso commented:
Renaming would be a divergance from conda-forge.

lorepirri commented:
> Renaming would be a divergance from conda-forge.

ok, no renaming

lorepirri commented:
> > Renaming would be a divergance from conda-forge.
> 
> ok, no renaming

right, conda-forge has something different under that name https://github.com/conda-forge/darts-feedstock

lorepirri committed:
only arm64, skip test_fit_on_residuals

lorepirri committed:
enable all valid builds, refactor recipe

- rename patch 002
- remove script files
- use inline script

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: clone/recipe/meta.yaml:127: missing_pip_check: For pypi packages, pip check should be present in the test commands
ERROR: clone/recipe/meta.yaml:127: missing_pip_check: For pypi packages, pip check should be present in the test commands
ERROR: clone/recipe/meta.yaml:127: missing_pip_check: For pypi packages, pip check should be present in the test commands
Errors were found

</details>

lorepirri committed:
enable all valid builds, refactor recipe

- rename patch 002

lorepirri committed:
Merge branch 'PKG-3090-udarts-8-all-0-26-0' of https://github.com/AnacondaRecipes/u8darts-feedstock into PKG-3090-udarts-8-all-0-26-0

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: clone/recipe/meta.yaml:127: missing_pip_check: For pypi packages, pip check should be present in the test commands
ERROR: clone/recipe/meta.yaml:127: missing_pip_check: For pypi packages, pip check should be present in the test commands
ERROR: clone/recipe/meta.yaml:127: missing_pip_check: For pypi packages, pip check should be present in the test commands
Errors were found

</details>

lorepirri committed:
restore scripts in file

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -66,159 +83,147 @@ outputs:
-  - name: u8darts-all
-    build:
-      noarch: python
+  - name: {{ name }}-all
```
I would keep the same package names as on conda-forge.

skupr-anaconda replied:
IMO, the conda-forge naming of the feedstock is partially incorrect. The existence of two PyPi projects darts/u8darts makes this naming issue more difficult. darts is more popular. I would say that u8darts is deprecated

skupr-anaconda replied:
We make darts and its synonym u8darts, and darts-all and u8darts-all subpackages

lorepirri replied:
Removed `darts`, we use `u8darts` and `u8darts-*`.

See comment https://github.com/AnacondaRecipes/u8darts-feedstock/pull/1#issuecomment-1774813629

skupr-anaconda commented:
> > > Renaming would be a divergance from conda-forge.
> > 
> > 
> > ok, no renaming
> 
> right, conda-forge has something different under that name https://github.com/conda-forge/darts-feedstock

The issue of the conda-forge that they are reserved the name `darts` for package `pt.darts` https://github.com/conda-forge/darts-feedstock/blob/main/recipe/meta.yaml. We can wait until they update both feedstock names.

skupr-anaconda reviewed (approved):
LGTM

M-Waszkiewicz-Anaconda reviewed (approved):
LGTM

lorepirri commented:
we could have the following names:

- `u8darts`
- `u8darts-all`
- `u8darts-torch`
- `u8darts-notorch`

removing the `darts` name, because there is https://github.com/khanrc/pt.darts named `darts` in:
- https://github.com/AnacondaRecipes/darts-feedstock
- https://github.com/conda-forge/darts-feedstock

even though on pipy both these names `darts` and `u8darts` point to: https://github.com/unit8co/darts
- https://pypi.org/project/u8darts/
- https://pypi.org/project/darts/

there are other darts:
https://github.com/quark0/darts

so probably prefixing the `darts` with the implementation is a better idea (e.g. `u8darts`, `pt-darts`, eventually `q0-darts`).

skupr-anaconda commented:
Thanks @lorepirri. Good finding. So our option is only `u8darts`-*

lorepirri committed:
use u8darts and u8darts-* names

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts-notorch": Output is missing script.
INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts-all": Output is missing script.
INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts-torch": Output is missing script.
ERROR: clone/recipe/meta.yaml:17: invalid_url: https://github.com/unit8co/u8darts/archive/refs/tags/0.26.0.tar.gz : Not reachable: 404
Errors were found

</details>

lorepirri committed:
restore github archive source

lorepirri committed:
format

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts-torch": Output is missing script.
INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts-all": Output is missing script.
INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts-notorch": Output is missing script.
All checks OK

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts-all": Output is missing script.
INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts-notorch": Output is missing script.
INFO: clone/recipe/meta.yaml:0: output_missing_script: output "u8darts-torch": Output is missing script.
All checks OK

</details>

lorepirri committed:
refer to the correct subpackage u8darts

lorepirri commented:
thank you @skupr-anaconda and @JeanChristopheMorinPerso for the support on this!
